functions/public/Invoke-WPFMicrowin.ps1

```diff
@@ -162,6 +162,35 @@ public class PowerManagement {
         Write-Host "Removing Appx Bloat"
         Remove-ProvisionedPackages
 
+        # Detect Windows 11 24H2 and add dependency to FileExp to prevent Explorer look from going back - thanks @WitherOrNot and @thecatontheceiling
+        if ((Test-CompatibleImage $imgVersion $([System.Version]::new(10,0,26100,1))) -eq $true)
+        {
+            try
+            {
+                if (Test-Path "$scratchDir\Windows\SystemApps\MicrosoftWindows.Client.FileExp_cw5n1h2txyewy\appxmanifest.xml" -PathType Leaf)
+                {
+                    # Found the culprit. Do the following:
+
+                    # 1. Take ownership of the file, from TrustedInstaller to Administrators
+                    takeown /F "$scratchDir\Windows\SystemApps\MicrosoftWindows.Client.FileExp_cw5n1h2txyewy\appxmanifest.xml" /A
+
+                    # 2. Set ACLs so that we can write to it
+                    icacls "$scratchDir\Windows\SystemApps\MicrosoftWindows.Client.FileExp_cw5n1h2txyewy\appxmanifest.xml" /grant "$(Get-LocalizedUsers -admins $true):(M)" | Out-Host
+
+                    # 3. Open the file and do the modification
+                    $appxManifest = Get-Content -Path "$scratchDir\Windows\SystemApps\MicrosoftWindows.Client.FileExp_cw5n1h2txyewy\appxmanifest.xml"
+                    $originalLine = $appxManifest[13]
+                    $dependency = "`n        <PackageDependency Name=`"Microsoft.WindowsAppRuntime.CBS`" MinVersion=`"1.0.0.0`" Publisher=`"CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US`" />"
+                    $appxManifest[13] = "$originalLine$dependency"
+                    Set-Content -Path "$scratchDir\Windows\SystemApps\MicrosoftWindows.Client.FileExp_cw5n1h2txyewy\appxmanifest.xml" -Value $appxManifest -Force -Encoding utf8
```

Comment:
I don't know if modifying the app manifest through line numbering is a good idea.. but I can't tell/know a better approach to fix this problem, other then pattern matching using RegEx.. which'll require a bit testing, and some headaches.

Reply:
I've used this method because I had used it earlier for DT (more on that [here](https://github.com/CodingWonders/DISMTools/blob/3cb2442727ffb7e337d6f6cc575a53a40c9fbf1c/Helpers/extps1/PE_Helper/PE_Helper.ps1#L1619-L1626)) and it works there.

I can't think of another way to insert a line on PowerShell either.

Reply:
Can I suggest this?
# Set file name
$File = '.\appxmanifest.xml'
echo Start

# Process lines of text from file and assign result to $NewContent variable
$NewContent = [System.IO.File]::ReadAllLines($File) |
    ForEach-Object {
        # Output the existing line to pipeline in any case
        $_

        # If line matches regex
        if($_ -match ('^' + [regex]::Escape('        <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.21340.0" MaxVersionTested="10.0.32767.0" />')))
        {
            # Add output additional line right after it
            '        <PackageDependency Name="Microsoft.WindowsAppRuntime.CBS" MinVersion="1.0.0.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" />'
        }
    }

# Write content of $NewContent varibale back to file
[System.IO.File]::WriteAllLines($File, $NewContent)

Reply:
Can I suggest this?
`# Set file name
$File = '.\appxmanifest.xml'
echo Start

# Process lines of text from file and assign result to $NewContent variable
$NewContent = [System.IO.File]::ReadAllLines($File) |
    ForEach-Object {
        # Output the existing line to pipeline in any case
        $_

        # If line matches regex
        if($_ -match ('^' + [regex]::Escape('        <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.21340.0" MaxVersionTested="10.0.32767.0" />')))
        {
            # Add output additional line right after it
            '        <PackageDependency Name="Microsoft.WindowsAppRuntime.CBS" MinVersion="1.0.0.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" />'
        }
    }

# Write content of $NewContent varibale back to file
[System.IO.File]::WriteAllLines($File, $NewContent)`

Reply:
Can I suggest this?

$File = '.\appxmanifest.xml'

$NewContent = [System.IO.File]::ReadAllLines($File) |
    ForEach-Object {
        $_
        if($_ -match ('^' + [regex]::Escape('        <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.21340.0" MaxVersionTested="10.0.32767.0" />')))
        {
            '        <PackageDependency Name="Microsoft.WindowsAppRuntime.CBS" MinVersion="1.0.0.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" />'
        }
    }

[System.IO.File]::WriteAllLines($File, $NewContent)

Reply:
`# Set file name
$File = '.\appxmanifest.xml'
echo Start

# Process lines of text from file and assign result to $NewContent variable
$NewContent = [System.IO.File]::ReadAllLines($File) |
    ForEach-Object {
        # Output the existing line to pipeline in any case
        $_

        # If line matches regex
        if($_ -match ('^' + [regex]::Escape('        <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.21340.0" MaxVersionTested="10.0.32767.0" />')))
        {
            # Add output additional line right after it
            '        <PackageDependency Name="Microsoft.WindowsAppRuntime.CBS" MinVersion="1.0.0.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" />'
        }
    }

# Write content of $NewContent varibale back to file
[System.IO.File]::WriteAllLines($File, $NewContent)`

Reply:
Can I suggest this?
```
# Set file name
$File = '.\appxmanifest.xml'

# Process lines of text from file and assign result to $NewContent variable
$NewContent = [System.IO.File]::ReadAllLines($File) |
    ForEach-Object {
        # Output the existing line to pipeline in any case
        $_

        # If line matches regex
        if($_ -match ('^' + [regex]::Escape('        <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.21340.0" MaxVersionTested="10.0.32767.0" />')))
        {
            # Add output additional line right after it
            '        <PackageDependency Name="Microsoft.WindowsAppRuntime.CBS" MinVersion="1.0.0.0" Publisher="CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US" />'
        }
    }

# Write content of $NewContent varibale back to file
[System.IO.File]::WriteAllLines($File, $NewContent)
```